### PR TITLE
[zh]Add 2021-12-16-StatefulSet-PVC-Auto-Deletion.md

### PR DESCRIPTION
**Pull request by zhangxyjlu**
Fixes:#32408

### CODE REVIEW

File: content/zh/blog/_posts/2021-12-16-StatefulSet-PVC-Auto-Deletion.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+in a StatefulSet. In both cases the policy can either be `Retain`, where the corresponding PVCs are
+not touched, or `Delete`, which means that PVCs are deleted. The deletion is done with a normal
+[object deletion](/docs/concepts/architecture/garbage-collection/), so that, for example, all
+retention policies for the underlying PV are respected.
+-->
+第一种情况是 StatefulSet 资源被删除时（这意味着所有副本也被删除），这由 `whenDeleted` 策略控制的。
+第二种情况是 StatefulSet 缩小时，即删除 StatefulSet 部分副本，这由 `whenScaled` 策略控制。
+在这两种情况下，策略即可以是 `Retain` 不涉及相应 PVCs 的改变，也可以是 `Delete` 即删除对应的 PVCs。
+删除是通过普通的[对象删除](/zh/docs/concepts/architecture/garbage-collection/)完成的，
+因此，底层 PV 的所有保留策略都应重视。
```
Comment (tengqm) [inline content/zh/blog/_posts/2021-12-16-StatefulSet-PVC-Auto-Deletion.md:95]:
```suggestion
因此，底层 PV 的所有保留策略都会被遵照执行。
```

File: content/zh/blog/_posts/2021-12-16-StatefulSet-PVC-Auto-Deletion.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+  * **`whenDeleted` and `whenScaled` are both `Delete`.** PVCs are deleted immediately when their
+    replica is no longer needed. Note this does not include when a Pod is deleted and a new version
+    rescheduled, for example when a node is drained and Pods need to migrate elsewhere. The PVC is
+    deleted only when the replica is no longer needed as signified by a scale-down or StatefulSet
+    deletion. This use case is for when data does not need to live beyond the life of its
+    replica. Perhaps the data is easily reconstructable and the cost savings of deleting unused PVCs
+    is more important than quick scale-up, or perhaps that when a new replica is created, any data
+    from a previous replica is not usable and must be reconstructed anyway.
+-->
+  * **`whenDeleted` 和 `whenScaled` 都是 `Delete`。** 当不再需要它们的副本时，PVCs 会立即被删除。
```
Comment (tengqm) [inline content/zh/blog/_posts/2021-12-16-StatefulSet-PVC-Auto-Deletion.md:138]:
```suggestion
  * **`whenDeleted` 和 `whenScaled` 都是 `Delete`。** 当其副本不再被需要时，PVCs 会立即被删除。
```

File: content/zh/blog/_posts/2021-12-16-StatefulSet-PVC-Auto-Deletion.md [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+    注意，这并不包括删除 Pod 和重新调度新版本的情况，例如当节点耗尽 Pod 需要迁移到别处时。
```
Comment (tengqm) [inline content/zh/blog/_posts/2021-12-16-StatefulSet-PVC-Auto-Deletion.md:139]:
```suggestion
    注意，这并不包括 Pod 被删除且有新版本被调度的情况，例如当节点被腾空而 Pod 需要迁移到别处时。
```

File: content/zh/blog/_posts/2021-12-16-StatefulSet-PVC-Auto-Deletion.md [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+    只有当副本不再需要时，如按比例缩小或删除 StatefulSet 时，才会删除 PVC。
```
Comment (tengqm) [inline content/zh/blog/_posts/2021-12-16-StatefulSet-PVC-Auto-Deletion.md:140]:
```suggestion
    只有当副本不再被需要时，如按比例缩小或删除 StatefulSet 时，才会删除 PVC。
```

File: content/zh/blog/_posts/2021-12-16-StatefulSet-PVC-Auto-Deletion.md [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+    此策略适用于数据生命周期小于副本生命周期的情况。即数据很容易重构，
```
Comment (tengqm) [inline content/zh/blog/_posts/2021-12-16-StatefulSet-PVC-Auto-Deletion.md:141]:
```suggestion
    此策略适用于数据生命周期短于副本生命周期的情况。即数据很容易重构，
```

File: content/zh/blog/_posts/2021-12-16-StatefulSet-PVC-Auto-Deletion.md [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+    如果需要使 Elasticsearch 系统脱机，可以通过临时删除 StatefulSet 来实现，
+    然后通过重新创建 StatefulSet 来恢复 Elasticsearch 集群。
+    保存 Elasticsearch 数据的 PVCs 不会被删除，新的副本将自动使用它们。
+  
+<!--
+  Visit the
+[documentation](docs/concepts/workloads/controllers/statefulset/#persistentvolumeclaim-policies) to
+see all the details.
+-->
+查阅[文档](docs/concepts/workloads/controllers/statefulset/#persistentvolumeclaim-policies)获取更多详细信息。
```
Comment (tengqm) [inline content/zh/blog/_posts/2021-12-16-StatefulSet-PVC-Auto-Deletion.md:171]:
```suggestion
查阅[文档](/zh/docs/concepts/workloads/controllers/statefulset/#persistentvolumeclaim-policies)
获取更多详细信息。
```

File: content/zh/blog/_posts/2021-12-16-StatefulSet-PVC-Auto-Deletion.md [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+测试一下，告诉我们你的体验！
+
+<!--
+I'm very curious to see if this owner reference mechanism works well in practice. For example, we
+realized there is no mechanism in Kubernetes for knowing who set a reference, so it’s possible that
+the StatefulSet controller may fight with custom controllers that set their own
+references. Fortunately, maintaining the existing retention behavior does not involve any new owner
+references, so default behavior will be compatible.
+-->
+我很好奇这个所有者引用机制在实践中是否有效。例如，我们意识到 Kubernetes 中没有可以知道谁设置了引用的机制，
```
Comment (tengqm) [inline content/zh/blog/_posts/2021-12-16-StatefulSet-PVC-Auto-Deletion.md:192]:
```suggestion
我很好奇这个属主引用机制在实践中是否有效。例如，我们意识到 Kubernetes 中没有可以知道谁设置了引用的机制，
```

**Comment by zhangxyjlu** [comment]
I have revised all the review comments. Thank you@[tengqm](https://github.com/tengqm)

**Comment by tengqm** [comment]
Thanks.
/label tide/merge-method-squash
/approve

**Pull request merged**